integration/spark-common/src/main/scala/org/apache/carbondata/spark/util/CommonUtil.scala

```diff
@@ -179,6 +193,68 @@ object CommonUtil {
     isValid
   }
 
+  def validateTypeConvert(desType: Option[String], value: String): Boolean = {
+    var flag = true
+    val result = desType match {
+      case Some("IntegerType") =>
+        DataTypeUtil.getDataBasedOnDataType(value.trim, DataType.INT)
+      case Some("StringType") =>
+        DataTypeUtil.getDataBasedOnDataType(value.trim, DataType.STRING)
```

Comment:
Please correct me if I am wrong....While validating the values you are trimming the values but in the actual range info can still contain the spaces as the object is not getting refilled with the trimmed values and therefore can lead to an inconsistent state. For e.g I have Int type column with rangeInfo = 1, 2 , 3,4...while parsing it will pass as trim function is used but original object still will have the same values...